packages/arb-bridge-eth/contracts/rollup/NodeFactory.sol

```diff
@@ -25,12 +25,13 @@ import "@openzeppelin/contracts/proxy/BeaconProxy.sol";
 import "@openzeppelin/contracts/proxy/UpgradeableBeacon.sol";
 
 contract NodeFactory is INodeFactory {
-    UpgradeableBeacon public beacon;
+    address public override beacon;
 
     constructor() public {
         address templateContract = address(new Node());
-        beacon = new UpgradeableBeacon(templateContract);
-        beacon.transferOwnership(msg.sender);
+        UpgradeableBeacon _beacon = new UpgradeableBeacon(templateContract);
+        _beacon.transferOwnership(msg.sender);
+        beacon = address(_beacon);
```

Comment:
To avoid this (we'd need a postUpgradeInit too) we could just check if the first node is upgraded instead of explicitly checking if the beacon is upgraded